File: Java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java

package com.adaptionsoft.games.trivia.runner;

import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Game;

public class GameRunner {

    public static void main(String[] args) {
        Random rand = new Random();
        main(rand);
    }

    static void main(Random rand) {
        Game aGame = new Game();

        aGame.add("Chet");
        aGame.add("Pat");
        aGame.add("Sue");

        boolean notAWinner;
        do {

            aGame.moveAndAsk(rand.nextInt(5) + 1);

            if (rand.nextInt(9) == 7) {
                notAWinner = aGame.wasWrongAnswered();
            } else {
                notAWinner = aGame.wasCorrectlyAnswered();
            }

        } while (notAWinner);

    }
}

File: Kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/runner/GameRunner.kt
package com.adaptionsoft.games.trivia.runner

import com.adaptionsoft.games.uglytrivia.Game
import com.adaptionsoft.games.uglytrivia.Logger
import java.util.*

object GameRunner {
    var notAWinner: Boolean = false
    val logger = Logger()
}

fun main(args: Array<String>) {
    val aGame = Game(GameRunner.logger)

    aGame.add("Chet")
    aGame.add("Pat")
    aGame.add("Sue")

    val rand = when (args.size) {
        1 -> Random(args[0].toLong())
        else -> Random()
    }

    do {

        aGame.roll(rand.nextInt(5) + 1)

        if (rand.nextInt(9) == 7) {
            GameRunner.notAWinner = aGame.wrongAnswer()
        } else {
            GameRunner.notAWinner = aGame.wasCorrectlyAnswered()
        }


    } while (GameRunner.notAWinner)

}
